tests/preprocessing/test_generator.py

```diff
@@ -76,16 +76,19 @@ def test_simple_filter(self):
             ]),
         ]
 
+        input_image = np.zeros((500, 500, 3))
+
         expected_annotations_group = [
             np.array([
                 [  0,   0, 10, 10],
             ]),
         ]
 
         simple_generator = SimpleGenerator(input_annotations_group)
+        annotations_group = simple_generator.load_annotations_group(simple_generator.groups[0])
         # expect a UserWarning
         with pytest.warns(UserWarning):
-            annotations_group = simple_generator.load_annotations_group(simple_generator.groups[0])
+            image_group, annotations_group = simple_generator.check_images_annotations([input_image], annotations_group, simple_generator.groups[0])
```

Comment:
Should we consider `check_image_annotations` to be a public interface? Or is it a private/protected implementation detail and/or extension point for subclasses?

In the latter case we should probably not test it directly but test the public interface instead. Or at the very least test that the public interface also filters properly.

Reply:
How do you propose to do that though? The public interface is through the `next()` function, but that returns anchor targets. Should we test on those anchor targets?

Reply:
As discussed in person: What we could do is have a dataset with two annotations with different labels, where one of the annotations has to be discarded by the filter. Then let the generator generate two values and assert that both of them have the label of the valid annotation.

This tests at the very least that the filtering is done.